src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,10 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                if (\PHP_VERSION_ID >= 70200) {
+                    throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please compile PHP with the --with-password-argon2 flag.');
+                }
+                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install libsodium via PECL.');
```

Comment:
I would keep the `or upgrade to PHP 7.2+` part of the message here (when you are on PHP 7.1, suggesting to upgrade to PHP 7.2 is still a good advice)